### ISSUE TITLE
EMBEDDED_ASSET without SRC gatsby-source-contentful/rich-text 

### ISSUE THREAD

**Issue by ilituchy** [contentful/rich-text#263]
Attempted to find a solution already out there, but could not.

Recently updated a blog project to current node / gatsby versions, and additionally updated gatsby-source-contentful to 15.13.0.

Now i am getting the RTF in a raw format instead of JSON, and therefore am using 

`import { renderRichText } from "gatsby-source-contentful/rich-text"`

rather than

`import { documentToReactComponents } from '@contentful/rich-text-react-renderer';`

However, I no longer have any information on the actually node that gets fed into [BLOCKS.EMBEDDED_ASSET]. I realize these are now stored on the references portion of the rich text rather than the raw portion. However, I am unable to figure out an efficient way to merge the references with the nodes / blocks within the raw data.

My graphql (14.5.8) query looks like this, where body is the rich text format field:
  
```
body {
      raw
      references {
        ... on ContentfulAsset {
             contentful_id
              fixed(width: 1600) {
                 width
                 height
                  src
                  srcSet
             }
        }
     }
  }
```

I have also tried 

```
body {
     raw
     references {
        contentful_id
        file {
           url
        }
     }
  }
```

to no avail. How are references supposed to be used, when the raw data is what is being rendered with the renderRichText method? How can I access the src of the asset to display within an img tag??

**Comment by dphogit** [comment]
Had this issue as well, in addition of your first query you need to include both `contentful_id` and `__typename` fields and that for some reason worked for me.

**Comment by ilituchy** [comment]
@dphogit thank you for the tip!! So is this how your reference section looks?

```
  references {
      ... on ContentfulAsset {
        contentful_id
        __typename
        fixed(width: 1600) {
          width
          height
          src
          srcSet
        }
      }
    }
```

And in terms of handling the Embedded asset within the options object, I currently have:

```
    [BLOCKS.EMBEDDED_ASSET]: node => <InlinedImage
      node={node}
    />,
```
with InlinedImage being a component I define above.

Is this what you have? I appreciate the help, the docs on this right now are bad / non-existent

**Comment by dphogit** [comment]
Yes the query is correct. The second part about the embedded asset  is correct and ultimately comes down to you what you want to as long as you provide the correct props etc. For example here would be a normal `<img />` tag for the query you provide:

``` JavaScript
[BLOCKS.EMBEDDED_ASSET]: node => (
          <img src={node.data.target.fixed.src} alt={...} /> // You can decide what alt you want to put in
      ),
```

Yeah I'm surprised the docs for this is not updated or explicitly says this considering the importance of embedded assets.
